wakuv2/waku.go

```diff
@@ -430,52 +429,19 @@ func (w *Waku) discoverAndConnectPeers() error {
 				continue
 			}
 
-			go w.identifyAndConnect(w.ctx, w.cfg.LightClient, *peerInfo)
+			go w.connect(*peerInfo, wps.Static)
 		}
 	}
 
 	return nil
 }
 
-func (w *Waku) identifyAndConnect(ctx context.Context, isLightClient bool, peerInfo peer.AddrInfo) {
-	ctx, cancel := context.WithTimeout(ctx, 7*time.Second)
-	defer cancel()
-
-	err := w.node.Host().Connect(ctx, peerInfo)
-	if err != nil {
-		w.logger.Error("could not connect to peer", zap.Any("peer", peerInfo), zap.Error(err))
-		return
-	}
-
-	conns := w.node.Host().Network().ConnsToPeer(peerInfo.ID)
-	if len(conns) == 0 {
-		return // No connection
-	}
-
-	select {
-	case <-w.ctx.Done():
-		return
-	case <-w.identifyService.IdentifyWait(conns[0]):
-		if isLightClient {
-			err = w.node.Host().Network().ClosePeer(peerInfo.ID)
-			if err != nil {
-				w.logger.Error("could not close connections to peer", zap.Stringer("peer", peerInfo.ID), zap.Error(err))
-			}
-			return
-		}
-
-		supportedProtocols, err := w.node.Host().Peerstore().SupportsProtocols(peerInfo.ID, relay.WakuRelayID_v200)
-		if err != nil {
-			w.logger.Error("could not obtain protocols", zap.Stringer("peer", peerInfo.ID), zap.Error(err))
-			return
-		}
-
-		if len(supportedProtocols) == 0 {
-			err = w.node.Host().Network().ClosePeer(peerInfo.ID)
-			if err != nil {
-				w.logger.Error("could not close connections to peer", zap.Stringer("peer", peerInfo.ID), zap.Error(err))
-			}
-		}
+func (w *Waku) connect(peerInfo peer.AddrInfo, origin wps.Origin) {
+	// Connection will be prunned eventually by the connection manager if needed
+	// The peer connector in go-waku uses Connect, so it will execute identify as part of its
+	connectedness := w.node.Host().Network().Connectedness(peerInfo.ID)
+	if connectedness == network.NotConnected || connectedness == network.CanConnect {
+		w.node.AddDiscoveredPeer(peerInfo.ID, peerInfo.Addrs, origin, nil, true)
```

Comment:
Wondering if pubsubTopics list should be empty or should we use defaultShardedPubsubTopic here?
Because if we are not connecting as relay, then pubsubTopic list won't be populated for the peers in peerstore and will lead to an issue in peer selection. 
Another option is to allow peer selection without pubsubTopics. 
WDYT @richard-ramos

Reply:
Makes sense. I'll introduce that change. Nodes for the network should still support the default shard pubsub topic so it makes sense to use it here